### DISCUSSION TITLE
invalid result from CGAL::IO::write_PLY()

### DISCUSSION THREAD

**Discussion by TomTheVercetti** [CGAL/cgal#7868] [Q&A]
Hi all,
for vizualisation purposes I need to export a mesh to a ply file. 
I use the following lines of code:
```
CGAL::IO::write_OBJ(OutputPath_PairwiseIntersection_obj, Mesh_3);
CGAL::IO::write_PLY(OutputPath_PairwiseIntersection_ply, Mesh_3);
```

where `OutputPath_PairwiseIntersection_obj` and `OutputPath_PairwiseIntersection_ply` are the output string paths.

the result of the `write_OBJ` is as expected:
![image](https://github.com/CGAL/cgal/assets/148688004/241431cd-ead2-4bc0-91cb-9b86031741f2)
 but when I use the same mesh for vizualisation, the output ply file is not valid: 
![image](https://github.com/CGAL/cgal/assets/148688004/938eac15-da16-4a62-9aca-541279bca322)
![image](https://github.com/CGAL/cgal/assets/148688004/4f833487-cd41-4849-9e8a-4850a34114c8)

Do you have a clue why this might be happening?

**Comment by sloriot** [comment]
The best way for us to help you is to provide a minimal example showing the problem that we can compile and run.

**Reply by TomTheVercetti** [reply]
@sloriot Sure!


here is an example, I have the same problem with this small program:
```
Mesh_3 me;
vertex_descriptor_3 v4 = me.add_vertex(Kernel::Point_3(0, 0, 0));
vertex_descriptor_3 v5 = me.add_vertex(Kernel::Point_3(1, 0, 0));
vertex_descriptor_3 v6 = me.add_vertex(Kernel::Point_3(1, 1, 0));
vertex_descriptor_3 v7 = me.add_vertex(Kernel::Point_3(0, 1, 0));
face_descriptor_2 f = me.add_face(v4, v5, v6, v7);
CGAL::IO::write_PLY("testMesh.ply", me);
```

**Comment by afabri** [comment]
I confirm that there is a problem and investigate.   A workaround is to store in Ascii instead of binary mode
```
CGAL::IO::write_PLY("testMesh.ply", me, "a comment", CGAL::parameters::use_binary_mode(false));
```

**Reply by afabri** [reply]
I wrote I can reproduce, but that is only true for the `CGAL::Exact_predicates_exact_constructions_kernel`.   What kernel do you use ?

**Reply by nimafo** [reply]
@afabri Thanks!
 I use the same kernel you mentioned.EPECK.

**Comment by stla** [comment]
There is an opened issue about this problem: https://github.com/CGAL/cgal/issues/7327

**Reply by afabri** [reply]
And a pull request #7874
